packages/completer/src/handler.ts

```diff
@@ -477,6 +511,98 @@ export namespace CompletionHandler {
     connector: IDataConnector<IReply, void, IRequest>;
   }
 
+  /**
+   * Wrapper object for ICompletionItem.
+   */
+  export interface ICompletionItems {
+    /**
+     * isIncomplete is true if this is a partial list of completion items.
+     */
+    isIncomplete: boolean;
+
+    /**
+     * Callback which can be used for actions such as fetching full documentation.
+     */
+    resolve?: (id: string) => void;
```

Comment:
Forgive my naivety, but shouldn't this function return something (e.g. the fully resolved `ICompletionItem`)? Or will it update the appropriate `ICompletionItem`?

Also, where does the provided `id` come from? I don't see such a field in `ICompletionItem`. If you envision it just updating an `ICompletionItem` in place, perhaps the interface can be `(item: ICompletionItem) => void`, to make it clear what's going on.

Reply:
Going with the latter option.